programs/ssl/ssl_client2.c

```diff
@@ -921,6 +921,22 @@ static int send_cb( void *ctx, unsigned char const *buf, size_t len )
     return( mbedtls_net_send( io_ctx->net, buf, len ) );
 }
 
+#if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+static int psk_cb(void * p_psk,  mbedtls_ssl_context *context, const unsigned char * identity, size_t  identity_len) {
+    (void)context;
+    (void)p_psk;
+
+    /* INFO: Cou can also configure the psk inside this callback. */
```

Comment:
Did I understand you correctly, the idea of the cli option `psk_callback=%%d` would be
to enable and to disable the callback in the `ssl_client2.c `?

To test the callback, I could pass the `psk` to the callback and install it from there, instead from main.

Reply:
Yes, precisely, the key should be installed from the callback instead of from main. The comment says it is possible, but we don't know that for sure unless we actually do it in a test.

Reply:
I added the option `psk_callback=%%d` to install the psk inside the callback function.